From 2c3b76987dc8feacdcbf59c397193f4e65adc77e Mon Sep 17 00:00:00 2001
From: JuriAbramov <openexr@dr-abramov.de>
Date: Mon, 19 Jan 2015 17:23:44 +0100
Subject: [PATCH] Fixed memory corruption / actual crashes on Window

Fixed memory corruption caused by missing assignment operator with non-trivial copy constructor logic.
FIxes crashes on Windows when "dwaa" or "dwab" codecs are used for saving files.
---
 OpenEXR/IlmImf/ImfDwaCompressorSimd.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/OpenEXR/IlmImf/ImfDwaCompressorSimd.h b/OpenEXR/IlmImf/ImfDwaCompressorSimd.h
index 93246f6..88ef0ad 100644
--- a/OpenEXR/IlmImf/ImfDwaCompressorSimd.h
+++ b/OpenEXR/IlmImf/ImfDwaCompressorSimd.h
@@ -91,6 +91,12 @@ class SimdAlignedBuffer64
             memcpy (_buffer, rhs._buffer, 64 * sizeof (T));
         }
 
+        SimdAlignedBuffer64 &operator=(const SimdAlignedBuffer64 &rhs)
+        {
+            memcpy (_buffer, rhs._buffer, 64 * sizeof (T));
+            return *this;
+        }
+
         ~SimdAlignedBuffer64 ()
         {
             EXRFreeAligned (_handle);
